freestyle/shared/src/main/scala/freestyle/internal/free.scala

```diff
@@ -201,7 +201,21 @@ private[internal] class Request(reqDef: Decl.Def, indexValue: Int) {
   private[this] val reqC    = Term.Name(req.value)
   private[this] val reqImpl = Term.Name(reqName)
 
-  val params: Seq[Term.Param] = reqDef.paramss.flatten
+  val cbtparams = for {
+    tps <- reqDef.tparams.filter(_.cbounds.nonEmpty)
+    cbound <- tps.cbounds
+  } yield Term.Param(Nil, Term.fresh(cboundPrefix), Some(Type.Apply(cbound, Seq(Type.Name(tps.name.value)))), None)
+
+  val tparams = reqDef.tparams.map {
+    case p if p.cbounds.nonEmpty => p.copy(cbounds = Nil)
+    case p => p
+  }
+
+  val params: Seq[Term.Param] = reqDef.paramss.flatten.map {
+    case p @ param"..$mods $paramname: $atpeopt = $expropt" /* if p.hasMod(mod"implicit")*/ =>
```

Comment:
There is `scalameta-contrib` lib that brings the `hasMod` method but it has also some transitive dependencies we don't want in the core for now.